File: apeu3.c
#include <stdio.h>
void completarMatrices(int matriz[2][3], int numMatrices);
void sumaMatriz(int matrizUno[2][3], int matrizDos[2][3], int suma[2][3]);
void restaMatriz(int matrizUno[2][3], int matrizDos[2][3], int resta[2][3]);
void multiplicacionMatriz(int matrizUno[2][3], int matrizDos[2][3], int multiplicacion[2][3]);
void mostrarResultado (int matriz[2][3]);

int main (){
    int matrizUno[2][3], matrizDos[2][3], suma[2][3], resta[2][3], multiplicacion[2][3], numMatrices;

    completarMatrices(matrizUno, 1);
    completarMatrices(matrizDos, 2);

    sumaMatriz(matrizUno, matrizDos, suma);
    restaMatriz(matrizUno, matrizDos, resta);
    multiplicacionMatriz(matrizUno, matrizDos, multiplicacion);

    
    return 0;
}

void completarMatrices(int matriz[2][3], int numMatrices){
    int i, j;
    printf ("---------------------------------------------------\n");
    printf ("Matriz %i\n", numMatrices);
    printf ("---------------------------------------------------\n");
    
    for (i = 0; i < 2; i++){
        for (j = 0; j < 3; j++){
            printf ("Ingresa la posicion %i, %i: ", i, j);
            scanf ("%i", &matriz[i][j]);
        }
    }
    

    for (i = 0; i < 2; i++){
        printf("[");
        for (j = 0; j < 3; j++){
            printf (" %i ", matriz[i][j]);
        }
        printf ("]\n");
    }
}

void sumaMatriz(int matrizUno[2][3], int matrizDos[2][3], int suma[2][3]){
    int i, j;
    for (i = 0; i < 2; i++){
        for (j = 0; j < 3; j++){
            suma[i][j] = matrizUno[i][j] + matrizDos[i][j];
        }
    }

    printf ("---------------------------------------------------\n");
    printf ("Suma de Matrices\n");
    printf ("---------------------------------------------------\n");

    mostrarResultado(suma);
}

void restaMatriz(int matrizUno[2][3], int matrizDos[2][3], int resta[2][3]){
    int i, j;
    for (i = 0; i < 2; i++){
        for (j = 0; j < 3; j++){
            resta[i][j] = matrizUno[i][j] - matrizDos[i][j];
        }
    }

    printf ("---------------------------------------------------\n");
    printf ("Resta de Matrices\n");
    printf ("---------------------------------------------------\n");

    mostrarResultado(resta); 
}

void multiplicacionMatriz(int matrizUno[2][3], int matrizDos[2][3], int multiplicacion[2][3]){
    int i, j;
    for (i = 0; i < 2; i++){
        for (j = 0; j < 3; j++){
            multiplicacion[i][j] = matrizUno[i][j] * matrizDos[i][j];
        }
    }

    printf ("---------------------------------------------------\n");
    printf ("Multiplicación de Matrices\n");
    printf ("---------------------------------------------------\n");

    mostrarResultado(multiplicacion);
}

void mostrarResultado (int matriz[2][3]){

    int i, j;

    for (i = 0; i < 2; i++){
        printf ("[");
        for (j = 0; j < 3; j++){
            printf (" %i ", matriz[i][j]);
        }
        printf ("]\n");
    }
    
}

File: apeu3.py
def completar_matrices(num_matrices):
    print("---------------------------------------------------")
    print(f"Matriz {num_matrices}")
    print("---------------------------------------------------")
    
    matriz = []
    for i in range(2):
        fila = []
        for j in range(3):
            val = int(input(f"Ingresa la posicion {i}, {j}: "))
            fila.append(val)
        matriz.append(fila)
        
    for i in range(2):
        print("[", end="")
        for j in range(3):
            print(f" {matriz[i][j]} ", end="")
        print("]")
        
    return matriz

def suma_matriz(matriz_uno, matriz_dos):
    suma = [[matriz_uno[i][j] + matriz_dos[i][j] for j in range(3)] for i in range(2)]
    print("---------------------------------------------------")
    print("Suma de Matrices")
    print("---------------------------------------------------")
    mostrar_resultado(suma)
    return suma

def resta_matriz(matriz_uno, matriz_dos):
    resta = [[matriz_uno[i][j] - matriz_dos[i][j] for j in range(3)] for i in range(2)]
    print("---------------------------------------------------")
    print("Resta de Matrices")
    print("---------------------------------------------------")
    mostrar_resultado(resta)
    return resta

def multiplicacion_matriz(matriz_uno, matriz_dos):
    multiplicacion = [[matriz_uno[i][j] * matriz_dos[i][j] for j in range(3)] for i in range(2)]
    print("---------------------------------------------------")
    print("Multiplicación de Matrices")
    print("---------------------------------------------------")
    mostrar_resultado(multiplicacion)
    return multiplicacion

def mostrar_resultado(matriz):
    for i in range(2):
        print("[", end="")
        for j in range(3):
            print(f" {matriz[i][j]} ", end="")
        print("]")

def main():
    matriz_uno = completar_matrices(1)
    matriz_dos = completar_matrices(2)

    suma_matriz(matriz_uno, matriz_dos)
    resta_matriz(matriz_uno, matriz_dos)
    multiplicacion_matriz(matriz_uno, matriz_dos)

if __name__ == "__main__":
    main()